app/server/src/db/sessionStore.ts

```diff
@@ -0,0 +1,21 @@
+import Redis from "ioredis";
+
+export class SessionStore {
+    private readonly _redis: Redis;
+
+    private readonly _sessionPrefix: string;
+
+    constructor(redis: Redis, course: string, app: string) {
+        this._redis = redis;
+        this._sessionPrefix = `${course}:${app}:sessions:`;
+    }
+
+    private sessionKey = (name: string) => `${this._sessionPrefix}${name}`;
+
+    async saveSession(id: string, data: any) {
+        await this._redis.pipeline()
+            .hset(this.sessionKey("time"), id, new Date(Date.now()).toISOString())
+            .hset(this.sessionKey("data"), id, JSON.stringify(data))
```

Comment:
Does it make sense to be doing the serialisation here, or upstream? The frontend will just be passing a string in here I'd have thought?

Reply:
Yes, you're right, it can just throw the whole body string at the db